test/mockupdb/test_handshake.py

```diff
@@ -229,6 +229,39 @@ def test_client_handshake_saslSupportedMechs(self):
                         future()
                     return
 
+    def test_client_handshake_saslSupportedMechs_unknown(self):
+        server = MockupDB()
+        server.run()
+        self.addCleanup(server.stop)
+
+        primary_response = OpReply(
+            "ismaster",
+            True,
+            minWireVersion=2,
+            maxWireVersion=MIN_SUPPORTED_WIRE_VERSION,
+            saslSupportedMechs=["SCRAM-SHA-256", "does_not_exist"],
+        )
+        client = MongoClient(
+            server.uri, authmechanism="PLAIN", username="username", password="password"
```

Comment:
I think this test may be flawed. A client only negotiates the auth mechanism if authmechanism is not supplied but this test sets `authmechanism="PLAIN"`. Instead it should create the client without authmechanism or set `authmechanism="DEFAULT"`.